sktime/transformations/tests/test_ytox.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+import pandas as pd
+import numpy as np
+from sktime.datasets import load_airline
+from sktime.transformations.compose import YtoX
+from sktime.transformations.series.exponent import ExponentTransformer
+from sktime.transformations.series.boxcox import BoxCoxTransformer
+
+
+@pytest.fixture
+def y_data():
+    """Fixture to create a sample dataset."""
+    y = load_airline()
+    return y
+
+
+def get_test_params():
+    """Provide test parameters for the YtoX class."""
+    # Return instances with different parameters
+    return [
+        {"ytox": YtoX(subset_index=False), "description": "YtoX without subset_index"},
+        {"ytox": YtoX(subset_index=True), "description": "YtoX with subset_index"},
+        {"ytox": YtoX(subset_index=False, transformer=ExponentTransformer(power=2)), "description": "YtoX with ExponentTransformer"},
+        {"ytox": YtoX(subset_index=False, transformer=BoxCoxTransformer()), "description": "YtoX with BoxCoxTransformer"},
+    ]
+
+
+def ensure_dataframe(output):
+    """Ensure the output is a DataFrame."""
+    if isinstance(output, pd.Series):
+        return output.to_frame()
+    return output
+
+
+@pytest.mark.parametrize("params", get_test_params())
+def test_ytox(params, y_data):
+    """Test YtoX with different parameters."""
+    ytox = params["ytox"]
+    description = params["description"]
+
+    X = pd.DataFrame(np.random.randn(len(y_data), 2), index=y_data.index)  # Example exogenous data
+
+    # Call fit first
+    ytox.fit(X, y_data)
+
+    transformed_y = ensure_dataframe(ytox.transform(X, y_data))
+
+    # Ensure both are DataFrames
+    if ytox.transformer is not None:
+        expected_y = ensure_dataframe(ytox.transformer.fit_transform(y_data))
+    else:
+        expected_y = ensure_dataframe(y_data)
+
+    try:
+        # Check if transformed data is the same as the original `y`
+        pd.testing.assert_frame_equal(transformed_y, expected_y)
+    except AssertionError as e:
+        print(f"\nError in {description}:")
+        print("\nTransformed y:\n", transformed_y.head())
+        print("\nExpected y:\n", expected_y.head())
+        print("\nError:\n", e)
+
+        # Perform a detailed comparison to find exact differences
+        diff = transformed_y.compare(expected_y)
+        print("\nDifferences between transformed and expected:\n", diff)
+
+        raise
+
+
+def test_ytox_with_transformer(y_data):
```

Comment:
@fkiraly should this test be parameterised with all compatible transformers, or we just test it on few transformers by initialising them inside the test ??